sdk/storage/Azure.Storage.Files/src/DirectoryClient.cs

```diff
@@ -1339,28 +1388,72 @@ private void SetNameFieldsIfNull()
-            string handleId = Constants.CloseAllHandles,
-            string marker = default,
+        public virtual async Task<int> ForceCloseAllHandlesAsync(
             bool? recursive = default,
             CancellationToken cancellationToken = default) =>
-            await ForceCloseHandlesInternal(
-                handleId,
-                marker,
+            await ForceCloseAllHandlesInternal(
                 recursive,
                 true, // async
                 cancellationToken)
                 .ConfigureAwait(false);
 
         /// <summary>
-        /// The <see cref="ForceCloseHandlesAsync"/> operation closes a handle or handles opened on a directory
+        /// The <see cref="ForceCloseAllHandlesAsync"/> operation closes all handles opened on a directory
+        /// or a file at the service. It optionally supports recursively closing handles on subresources
+        /// when the resource is a directory.
+        ///
+        /// This API is intended to be used alongside <see cref="GetHandlesAsync"/> to force close handles that
+        /// block operations. These handles may have leaked or been lost track of by
+        /// SMB clients. The API has client-side impact on the handle being closed, including user visible
+        /// errors due to failed attempts to read or write files. This API is not intended for use as a replacement
+        /// or alternative for SMB close.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/force-close-handles"/>.
+        /// </summary>
+        /// <param name="recursive">
+        /// Optional. A boolean value that specifies if the operation should also apply to the files and subdirectories of the directory specified.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// The number of handles closed.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<int> ForceCloseAllHandlesInternal(
+            bool? recursive,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            int handlesClosed = 0;
+            string marker = null;
+            do
+            {
+                Response<StorageClosedHandlesSegment> response =
+                    await ForceCloseHandlesInternal(Constants.CloseAllHandles, marker, recursive, async, cancellationToken).ConfigureAwait(false);
+                marker = response.Value.Marker;
+                handlesClosed += response.Value.NumberOfHandlesClosed;
+
```

Comment:
Nit - extra line

Reply:
I actually added that intentionally to separate the return a bit as it was kind of hard to read with the while condition right above it.